src/shims/windows/fs.rs

```diff
@@ -152,7 +153,7 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
 
         let file_name =
             String::from_utf16_lossy(&this.read_wide_str(this.read_pointer(file_name)?)?);
-        let file_name = Path::new(&file_name);
+        let file_name = local_path(&file_name);
```

Comment:
This does not look right, you should not have to write any new logic here.

Did you try `this.read_path_from_wide_str`? That is the intended function for cases like this.

Also, why are you going through `String`?  `PathBuf` should generally be used for host file names, and the Rust file APIs typically support that.

Reply:
I added this to fix test failures on Linux and Mac.

Windows file names are (at least sometimes) prefixed with `\\?\` to make them able to go over the old max path size, or may include the drive like `C:\`. This prefix must be stripped before the path is passed to the actual local APIs, or else it will cause errors on unix (since PathBuf assumes paths are local, AFAIK). The other way around (emulating unix on Windows) doesn't need this because Windows gracefully handles `/` as a secondary separator, and a starting one as root of the current drive.

Reply:
`read_path_from_wide_str` handled all that already. At least, it should.

Reply:
Ah, cool. Sorry I missed that, I'll swap to that function